.github/workflows/build.yml

```diff
@@ -0,0 +1,24 @@
+name: Java CI
+
+on: [push]
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v3
+
+      - name: Set up JDK 17
+        uses: actions/setup-java@v3
+        with:
+          java-version: '17'
+          distribution: 'adopt'
+
+      - name: Build with Maven
+        run: mvn --batch-mode --update-snapshots package
```

Comment:
let's make this `verify` instead of `package`, so that if/when we have integration tests, they run too?

Reply:
verify target does not work.
There are some (I suppose) surefire tests, but they seem to fail in github action, probably due to missing display.
It would be nice to have integration tests working, but probably it requires more effort than reasonable.

```
2022-12-13T23:15:12.9030979Z [ERROR] Failed to execute goal org.eclipse.tycho:tycho-surefire-plugin:3.0.0:test (default-test) on project com.gwtplugins.gdt.eclipse.core.test: An unexpected error occurred while launching the test runtime (process returned error code 13). The process logfile /home/runner/work/gwt-eclipse-plugin/gwt-eclipse-plugin/plugins/com.gwtplugins.gdt.eclipse.core.test/target/work/data/.metadata/.log might contain further details. Command-line used to launch the sub-process was /opt/hostedtoolcache/Java_Adopt_jdk/17.0.5-8/x64/bin/java -Dosgi.noShutdown=false -Dosgi.os=linux -Dosgi.ws=gtk -Dosgi.arch=x86_64 -Dorg.eclipse.swtbot.search.timeout=50000 -Xms40m -Xmx1G -Djava.awt.headless=true -Dosgi.clean=true -jar /home/runner/.m2/repository/p2/osgi/bundle/org.eclipse.equinox.launcher/1.6.400.v20210924-0641/org.eclipse.equinox.launcher-1.6.400.v20210924-0641.jar -data /home/runner/work/gwt-eclipse-plugin/gwt-eclipse-plugin/plugins/com.gwtplugins.gdt.eclipse.core.test/target/work/data -install /home/runner/work/gwt-eclipse-plugin/gwt-eclipse-plugin/plugins/com.gwtplugins.gdt.eclipse.core.test/target/work -configuration /home/runner/work/gwt-eclipse-plugin/gwt-eclipse-plugin/plugins/com.gwtplugins.gdt.eclipse.core.test/target/work/configuration -application org.eclipse.tycho.surefire.osgibooter.uitest -testproperties /home/runner/work/gwt-eclipse-plugin/gwt-eclipse-plugin/plugins/com.gwtplugins.gdt.eclipse.core.test/target/surefire.properties -testApplication org.eclipse.ui.ide.workbench -product org.eclipse.sdk.ide -nouithread in working directory /home/runner/work/gwt-eclipse-plugin/gwt-eclipse-plugin/plugins/com.gwtplugins.gdt.eclipse.core.test -> [Help 1]
```

Reply:
Setting up a display shouldn't be too hard, either there is a github action out there for xvfb or we can just build inside a docker image that provides it instead (selenium chrome/firefox images do this, for example).

If there are only a few tests that are incorrectly failing, perhaps we can just ignore them for the time being, but still get benefits from others? If we aren't running tests, how confident can we be in this continuous integration setup?